src/main/java/com/onwelo/tinvalidator/validator/NLValidator.java

```diff
@@ -0,0 +1,25 @@
+package com.onwelo.tinvalidator.validator;
+
+import java.util.regex.Pattern;
+
+public class NLValidator implements Validator {
+
+    private static final Pattern PATTERN = Pattern.compile("[0-9]{9}B[0-9]{2}");
+
+    public Pattern getPattern(){
+            return PATTERN;
+    }
+
+    public boolean computeControlSum(String tin) {
+        int[] weights = {9, 8, 7, 6, 5, 4, 3, 2};
+        try {
+            int sum = 0;
+            for (int i = 0; i < weights.length; i++) {
+                sum += Integer.parseInt(tin.substring(i, i + 1)) * weights[i];
+            }
+            return (sum % 11) == Integer.parseInt(tin.substring(8, 9));
```

Comment:
`If A2 = 10 then number is invalid else C9 = A2`
Brakuje nam sprawdzenia `(czy sum % 11) != 10`

Reply:
I jest jeszcze drugi niesprawdzony warunek `[C11 C12] | >00`.

Reply:
Co do pierwszego komentarza, jedna cyfra nigdy nie będzie równa 10, więc ten warunek jest zachowany. Drugi zaraz będzie.